src/Illuminate/Encryption/Encrypter.php

```diff
@@ -184,12 +194,11 @@ public function decryptString($payload)
      *
      * @param  string  $iv
      * @param  mixed  $value
-     * @param  string  $tag
      * @return string
      */
-    protected function hash($iv, $value, $tag = '')
+    protected function hash($iv, $value)
```

Comment:
This is technically a breaking change.

Reply:
Good catch, should I add the parameter back, but simply keep it unused?

Reply:
I think it's important to reinstate the previous behavior of this method because otherwise some implementations will definitely break. Can we work around it in the method itself?

Reply:
It was technically a breaking change to add this new parameter to this method, but that didn't prevent it from being merged into 8.x only a couple of weeks ago so if the same criteria apply in this case it shouldn't be a problem to remove this parameter on a patch release too (otherwise there's no consistent policy).

Reply:
@X-Coder264 It wasn't breaking then because it has a default value, right?

I have reverted this now, and the old behavior of `hash` and `validMac`, should cause any problems.

Reply:
> It was technically a breaking change to add this new parameter to this method, but that didn't prevent it from being merged into 8.x only a couple of weeks ago 

Where did this happen?

Reply:
> (otherwise there's no consistent policy)

Sometimes things slip through or exceptions are made. These rules aren't set in stone and we have to decide ourselves what's best at the time.

Reply:
@Krisell The default value unfortunatelly does not help.

http://sandbox.onlinephpfunctions.com/code/4e64d1fbca122502dbb1c637d0e9e490561c9fb2

Warning:  Declaration of Bar::hash($iv, $value) should be compatible with Foo::hash($iv, $value, $tag = '') in on line 13

Reply:
>  The default value unfortunately does not help with backwards compatibility.

@X-Coder264 I'm not sure what you're getting at? All changes related to the method signature were reverted.

Reply:
> > It was technically a breaking change to add this new parameter to this method, but that didn't prevent it from being merged into 8.x only a couple of weeks ago
> 
> Where did this happen?

@driesvints  https://github.com/laravel/framework/pull/38190/files

`$tag` was added here and that PR was merged into 8.x two weeks ago.

> Sometimes things slip through or exceptions are made. These rules aren't set in stone and we have to decide ourselves what's best at the time.

Yeah, I know, but a LOT of methods in Laravel are `protected` and there's no final classes. I mean not using `final` classes is fine, but there's just so much `protected` stuff that a lot of changes become breaking just because the thing wasn't `private` from the start.

Reply:
> $tag was added here and that PR was merged into 8.x two weeks ago.

Okay, *now* I see what you mean. I did not know about this 😓 

@Krisell in that case it's fine if you want to remove the `$tag` param again. Sorry about that...

> Yeah, I know, but a LOT of methods in Laravel are protected and there's no final classes. I mean not using final classes is fine, but there's just so much protected stuff that a lot of changes become breaking just because the thing wasn't private from the start.

@X-Coder264 I realize that but that's just how things are unfortunately.

I'm gonna close this convo now. Let's keep this thread focused on the PR at hand.